app/views/transactions/create.js.erb

```diff
@@ -1,10 +1,23 @@
 <% if @transaction.persisted? %>
-  $("<%=j render('transactions/transaction', transaction: @transaction) %>").hide().prependTo(".transactions").fadeIn(1000);
-  $("#new_transaction").replaceWith("<%=j render('transactions/form', transaction: Transaction.new(category_id: params[:transaction][:fixed_category_id]), fixed_category_id: params[:transaction][:fixed_category_id]) %>");
-  scrollTo($("#<%= dom_id(@transaction) %>"));
-  $("#sidebar").replaceWith("<%= j render('home/sidebar') %>");
-  $("#total_balance").replaceWith("<%= j render('shared/layout/total_balance') %>");
+  if ($('.transactions').size() == 0) {
+    location.reload();
+  }
+  else {
+    $("<%=j render('transactions/transaction', transaction: @transaction) %>").hide().prependTo(".transactions").fadeIn(1000);
+    if (document.URL == "<%= root_url %>" && $('.transactions tbody tr td[colspan]').size() == 0) {
```

Comment:
@Gataniel I think we don't need this feature (when I see line under my new transaction). We need this feature only for new transactions if I have not seen them before, but when I create my transaction - I know about it and it should not be marked as unread. I suggest to use more deep colors for unread transactions, it will be easier to implement, all you need is just check if `transaction.created_at > current_member.last_viewed_at` in `app/views/transactions/_transaction.html.slim` file.